app/services/hyrax/collections/collection_member_service.rb

```diff
@@ -67,6 +67,96 @@ def available_member_work_ids
         end
         response
       end
+
+      class << self
+        # Check if a work or collection is already a member of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param member [Hyrax::Resource] the child collection and/or child work to check
+        # @returns [Boolean] true if already in the member set; otherwise, false
+        def member?(collection:, member:)
+          member.member_of_collection_ids.include? collection.id
+        end
+
+        # Add works and/or collections as members of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param new_member_ids [Enumerable<Valkyrie::ID>] the ids of the new child collections and/or child works
+        # @returns [Enumerable<Hyrax::Resource>] updated member resources
+        def add_members_by_ids(collection:, new_member_ids:, user:)
+          new_members = Hyrax.query_service.find_many_by_ids(ids: new_member_ids)
+          add_members(collection: collection, new_members: new_members, user: user)
+        end
+
+        # Add works and/or collections as members of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param new_members [Enumerable<Hyrax::Resource>] the new child collections and/or child works
+        # @returns [Enumerable<Hyrax::Resource>] updated member resources
+        def add_members(collection:, new_members:, user:)
+          new_members.map { |new_member| add_member(collection: collection, new_member: new_member, user: user) }
+        end
+
+        # Add a work or collection as a member of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param new_member_id [Valkyrie::ID] the id of the new child collection or child work
+        # @returns [Hyrax::Resource] updated member resource
+        def add_member_by_id(collection:, new_member_id:, user:)
+          new_member = Hyrax.query_service.find_by(id: new_member_id)
+          add_member(collection: collection, new_member: new_member, user: user)
+        end
+
+        # Add a work or collection as a member of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param new_member [Hyrax::Resource] the new child collection or child work
+        # @returns [Hyrax::Resource] updated member resource
+        def add_member(collection:, new_member:, user:)
+          message = Hyrax::MultipleMembershipChecker.new(item: new_member).check(collection_ids: collection.id, include_current_members: true)
+          if message
+            new_member.errors.add(:collections, message)
```

Comment:
where does the `#errors` method come from. this looks like an `ActiveModel` thing, and i'm surprised it's working here.

my intuition is that this method should `raise` something in the case that it refuses to add the member. i'm not sure a valkyrie user will know to check `errors` to determine if this method succeeded or not (would they need to check it every time?).

Reply:
This is part of the MultipleMembershipChecker.  I can change it to raise the error and see if anything breaks.

Reply:
FYI... This code came from [models/concerns/hyrax/collection_behavior.rb #add_member_objects](https://github.com/samvera/hyrax/blob/v3.0.2/app/models/concerns/hyrax/collection_behavior.rb#L61-L75).

Reply:
i think that code is handling only ActiveFedora objects.

Reply:
i'm still pretty sure this will fail if the member passed in isn't generated by Wings. e.g. if it's a `Monograph`.

Reply:
DONE: Change to raise an exception.